controllers/serviceinstance_controller.go

```diff
@@ -98,11 +104,10 @@ func (r *ServiceInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		instance, err := r.getInstanceForRecovery(smClient, serviceInstance, log)
 		if err != nil {
 			log.Error(err, "failed to check instance recovery")
-			setFailureConditions(smTypes.CREATE, err.Error(), serviceInstance)
-			if err := r.updateStatusWithRetries(ctx, serviceInstance, log); err != nil {
-				return ctrl.Result{}, err
+			if isTransientError(err, log) {
+				return r.markAsTransientError(ctx, smTypes.CREATE, err.Error(), serviceInstance, log)
 			}
-			return ctrl.Result{Requeue: true, RequeueAfter: r.Config.SyncPeriod}, nil
+			return r.markAsNonTransientError(ctx, smTypes.CREATE, err.Error(), serviceInstance, log)
 		}
 		if instance != nil {
```

Comment:
it can be update as well, but I think we have no way to know it

Reply:
what do you mean? 
if the error is non-transient there is no need to requeue